### DISCUSSION TITLE
Calculating TVL for the pool contracts

### DISCUSSION THREAD

**Discussion by 0xmonkeyking** [DefiLlama/DefiLlama-Adapters#11057] [Q&A] [answered]
How can I calculate TVL for pool contracts in the protocol? Which functions are useful?

**Comment by 0xtomoon** [comment, marked as answer]
@0xmonkeyking 
There are several functions in the `projects/helper/unWrapLPs.js`
You can use `sumTokens2` function to get tokens balance from pool contracts.

https://github.com/DefiLlama/DefiLlama-Adapters/blob/main/projects/stargatefinance-v2/index.js#L91
```
const createTvlFunction = (pools) => {
  return async (api) => {
    const tokens = await api.multiCall({ abi: 'address:token', calls: pools, })
    return sumTokens2({ api, tokensAndOwners2: [tokens, pools] })
  };
};
```
Here is one example which uses sumTokens2. Hope it answers your question.
